core/api/src/app/wallets/update-single-pending-invoice.ts

```diff
@@ -0,0 +1,336 @@
+// Package-internal method, not re-exported from index.ts but used in tests
+
+import { getTransactionForWalletByJournalId } from "./get-transaction-by-journal-id"
+
+import { declineHeldInvoice } from "./decline-single-pending-invoice"
+
+import { removeDeviceTokens } from "@/app/users/remove-device-tokens"
+import { getCurrentPriceAsDisplayPriceRatio, usdFromBtcMidPriceFn } from "@/app/prices"
+
+import { CouldNotFindError, CouldNotFindWalletInvoiceError } from "@/domain/errors"
+import { checkedToSats } from "@/domain/bitcoin"
+import { DisplayAmountsConverter } from "@/domain/fiat"
+import { InvoiceNotFoundError } from "@/domain/bitcoin/lightning"
+import { paymentAmountFromNumber, WalletCurrency } from "@/domain/shared"
+import { WalletInvoiceReceiver } from "@/domain/wallet-invoices/wallet-invoice-receiver"
+import { DeviceTokensNotRegisteredNotificationsServiceError } from "@/domain/notifications"
+
+import {
+  addAttributesToCurrentSpan,
+  recordExceptionInCurrentSpan,
+  wrapAsyncToRunInSpan,
+} from "@/services/tracing"
+import {
+  AccountsRepository,
+  WalletInvoicesRepository,
+  WalletsRepository,
+  UsersRepository,
+} from "@/services/mongoose"
+import { LndService } from "@/services/lnd"
+import { LockService } from "@/services/lock"
+import * as LedgerFacade from "@/services/ledger/facade"
+import { DealerPriceService } from "@/services/dealer-price"
+import { NotificationsService } from "@/services/notifications"
+
+import { toDisplayBaseAmount } from "@/domain/payments"
+
+const lockedUpdatePendingInvoiceSteps = async ({
+  paymentHash,
+  recipientWalletId,
+  receivedBtc,
+  description,
+  isSettledInLnd,
```

Comment:
Should `isSettledInLnd` not be passed across the lock because it is stateful and can change by the time the lock is acquired.

Reply:
Hm I think in this case it's probably fine since `lndService.settleInvoice` is idempotent